# HG changeset patch
# User Rik <rik@octave.org>
# Date 1547436819 28800
#      Sun Jan 13 19:33:39 2019 -0800
# Node ID 0ba9ecd3814b64b0e1e5d12456eb3979e0ce4cc8
# Parent  8c3e727c44b531c93924e79608d8a886f376e6fd
# Parent  e590d8ad9ed01e021940e0f9f4dc5f63188851b8
maint: merge stable to default.

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -1414,37 +1414,45 @@ if test $ENABLE_JIT = yes; then
     AC_CHECK_HEADER([llvm/Support/TargetSelect.h], [warn_llvm=""])
 
     have_function_h=no
     AC_CHECK_HEADERS([llvm/IR/Function.h llvm/Function.h],
                      [have_function_h=yes; break])
     if test $have_function_h = no; then
       warn_llvm="Missing LLVM file Function.h.  JIT compiler is disabled."
     fi
+
     have_irbuilder_h=no
     AC_CHECK_HEADERS([llvm/Support/IRBuilder.h llvm/IR/IRBuilder.h \
                       llvm/IRBuilder.h], [have_irbuilder_h=yes; break])
     if test $have_irbuilder_h = no; then
       warn_llvm="Missing LLVM file IRBuilder.h.  JIT compiler is disabled."
     fi
+
     have_llvm_data_h=no
     AC_CHECK_HEADERS([llvm/Target/TargetData.h llvm/IR/DataLayout.h \
                       llvm/DataLayout.h], [have_llvm_data_h=yes; break])
     if test $have_llvm_data_h = no; then
       warn_llvm="Missing LLVM file TargetData.h.  JIT compiler is disabled."
     fi
 
     AC_CHECK_HEADERS([llvm/IR/Verifier.h])
     AC_CHECK_HEADERS([llvm/Analysis/BasicAliasAnalysis.h])
+    AC_CHECK_HEADERS([llvm/Transforms/Scalar/GVN.h])
+    AC_CHECK_HEADERS([llvm/Bitcode/ReaderWriter.h])
 
     OCTAVE_LLVM_FUNCTION_ADDATTRIBUTE_API
     OCTAVE_LLVM_FUNCTION_ADDFNATTR_API
     OCTAVE_LLVM_CALLINST_ADDATTRIBUTE_API
     OCTAVE_LLVM_RAW_FD_OSTREAM_API
     OCTAVE_LLVM_LEGACY_PASSMANAGER_API
+    OCTAVE_LLVM_IRBUILDER_API
+    OCTAVE_LLVM_HAS_CREATEALWAYSINLINERPASS
+    OCTAVE_LLVM_IRBUILDER_CREATECONSTINBOUNDSGEP1_32_API
+
     AC_LANG_POP(C++)
     CPPFLAGS="$save_CPPFLAGS"
     LDFLAGS="$save_LDFLAGS"
   fi
 
   if test -z "$warn_llvm"; then
     AC_DEFINE(HAVE_LLVM, 1, [Define to 1 if LLVM is available.])
   else
diff --git a/libgui/languages/it_IT.ts b/libgui/languages/it_IT.ts
--- a/libgui/languages/it_IT.ts
+++ b/libgui/languages/it_IT.ts
@@ -100,17 +100,17 @@
     <message>
         <location line="+5"/>
         <source>Select All</source>
         <translation>Seleziona tutto</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Run Selection</source>
-        <translation type="unfinished">Esegui selezione</translation>
+        <translation>Esegui selezione</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Clear Window</source>
         <translation>Pulisci finestra</translation>
     </message>
     <message>
         <location filename="../qterminal/libqterminal/QTerminal.cc" line="+49"/>
@@ -126,675 +126,675 @@
         <translation>selezione copiata negli appunti</translation>
     </message>
 </context>
 <context>
     <name>QsciLexerBash</name>
     <message>
         <location filename="../../../qsci/qscilexerbash.cpp" line="+208"/>
         <source>Default</source>
-        <translation type="unfinished">Default</translation>
+        <translation>Default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Error</source>
-        <translation type="unfinished">Errore</translation>
+        <translation>Errore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Number</source>
-        <translation type="unfinished"></translation>
+        <translation>Numero</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici doppi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Single-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici singoli</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Operator</source>
-        <translation type="unfinished"></translation>
+        <translation>Operatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Identifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Identificatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Scalar</source>
-        <translation type="unfinished"></translation>
+        <translation>Scalare</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Parameter expansion</source>
-        <translation type="unfinished"></translation>
+        <translation>Espansione parametro</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Backticks</source>
-        <translation type="unfinished"></translation>
+        <translation>Apici inversi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Here document delimiter</source>
-        <translation type="unfinished"></translation>
+        <translation>Delimitatore di here document</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Single-quoted here document</source>
-        <translation type="unfinished"></translation>
+        <translation>Here document con apici singoli</translation>
     </message>
 </context>
 <context>
     <name>QsciLexerBatch</name>
     <message>
         <location filename="../../../qsci/qscilexerbatch.cpp" line="+179"/>
         <source>Default</source>
-        <translation type="unfinished">Default</translation>
+        <translation>Default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Label</source>
-        <translation type="unfinished"></translation>
+        <translation>Etichetta</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Hide command character</source>
-        <translation type="unfinished"></translation>
+        <translation>Nascondi carattere comando</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>External command</source>
-        <translation type="unfinished"></translation>
+        <translation>Comando esterno</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Variable</source>
-        <translation type="unfinished"></translation>
+        <translation>Variabile</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Operator</source>
-        <translation type="unfinished"></translation>
+        <translation>Operatore</translation>
     </message>
 </context>
 <context>
     <name>QsciLexerCPP</name>
     <message>
         <location filename="../../../qsci/qscilexercpp.cpp" line="+352"/>
         <source>Default</source>
-        <translation type="unfinished">Default</translation>
+        <translation>Default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive default</source>
-        <translation type="unfinished"></translation>
+        <translation>Default inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>C comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive C comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>C++ comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C++</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive C++ comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C++ inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>JavaDoc style C comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C in stile JavaDoc</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive JavaDoc style C comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C inattivo in stile JavaDoc</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Number</source>
-        <translation type="unfinished"></translation>
+        <translation>Numero</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive number</source>
-        <translation type="unfinished"></translation>
+        <translation>Numero inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici doppi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive double-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici doppi inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Single-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici singoli</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive single-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici singoli inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>IDL UUID</source>
-        <translation type="unfinished"></translation>
+        <translation>IDL UUID</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive IDL UUID</source>
-        <translation type="unfinished"></translation>
+        <translation>IDL UUID inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Pre-processor block</source>
-        <translation type="unfinished"></translation>
+        <translation>Blocco del preprocessore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive pre-processor block</source>
-        <translation type="unfinished"></translation>
+        <translation>Blocco del preprocessore inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Operator</source>
-        <translation type="unfinished"></translation>
+        <translation>Operatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive operator</source>
-        <translation type="unfinished"></translation>
+        <translation>Operatore inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Identifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Identificatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive identifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Identificatore inattivo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Unclosed string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa aperta</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive unclosed string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa aperta inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>C# verbatim string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa verbatim C#</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive C# verbatim string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa verbatim C# inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>JavaScript regular expression</source>
-        <translation type="unfinished"></translation>
+        <translation>Espressione regolare JavaScript</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive JavaScript regular expression</source>
-        <translation type="unfinished"></translation>
+        <translation>Espressione regolare JavaScript inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>JavaDoc style C++ comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C++ in stile JavaDoc</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive JavaDoc style C++ comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento C++ inattivo in stile JavaDoc</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Secondary keywords and identifiers</source>
-        <translation type="unfinished"></translation>
+        <translation>Parole chiave e identificatori secondari</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive secondary keywords and identifiers</source>
-        <translation type="unfinished"></translation>
+        <translation>Parole chiave e identificatori secondari inattivi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>JavaDoc keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave JavaDoc</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive JavaDoc keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave JavaDoc inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>JavaDoc keyword error</source>
-        <translation type="unfinished"></translation>
+        <translation>Errore parola chiave JavaDoc</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive JavaDoc keyword error</source>
-        <translation type="unfinished"></translation>
+        <translation>Errore parola chiave JavaDoc inattiva</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Global classes and typedefs</source>
-        <translation type="unfinished"></translation>
+        <translation>Classi e typedef globali</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive global classes and typedefs</source>
-        <translation type="unfinished"></translation>
+        <translation>Classi e typedef globali inattive</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>C++ raw string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa raw C++</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Inactive C++ raw string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa raw C++ inattiva</translation>
     </message>
 </context>
 <context>
     <name>QsciLexerDiff</name>
     <message>
         <location filename="../../../qsci/qscilexerdiff.cpp" line="+107"/>
         <source>Default</source>
-        <translation type="unfinished">Default</translation>
+        <translation>Default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Command</source>
-        <translation type="unfinished"></translation>
+        <translation>Comando</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Header</source>
-        <translation type="unfinished"></translation>
+        <translation>Intestazione</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Position</source>
-        <translation type="unfinished">Posizione</translation>
+        <translation>Posizione</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Removed line</source>
-        <translation type="unfinished"></translation>
+        <translation>Linea rimossa</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Added line</source>
-        <translation type="unfinished"></translation>
+        <translation>Linea aggiunta</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Changed line</source>
-        <translation type="unfinished"></translation>
+        <translation>Linea modificata</translation>
     </message>
 </context>
 <context>
     <name>QsciLexerMatlab</name>
     <message>
         <location filename="../../../qsci/qscilexermatlab.cpp" line="+138"/>
         <source>Default</source>
-        <translation type="unfinished">Default</translation>
+        <translation>Default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Command</source>
-        <translation type="unfinished"></translation>
+        <translation>Comando</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Number</source>
-        <translation type="unfinished"></translation>
+        <translation>Numero</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Single-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici singoli</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Operator</source>
-        <translation type="unfinished"></translation>
+        <translation>Operatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Identifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Identificatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici doppi</translation>
     </message>
 </context>
 <context>
     <name>QsciLexerPerl</name>
     <message>
         <location filename="../../../qsci/qscilexerperl.cpp" line="+333"/>
         <source>Default</source>
-        <translation type="unfinished">Default</translation>
+        <translation>Default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Error</source>
-        <translation type="unfinished">Errore</translation>
+        <translation>Errore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Comment</source>
-        <translation type="unfinished"></translation>
+        <translation>Commento</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>POD</source>
-        <translation type="unfinished"></translation>
+        <translation>POD</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Number</source>
-        <translation type="unfinished"></translation>
+        <translation>Numero</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Keyword</source>
-        <translation type="unfinished"></translation>
+        <translation>Parola chiave</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici doppi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Single-quoted string</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici singoli</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Operator</source>
-        <translation type="unfinished"></translation>
+        <translation>Operatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Identifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Identificatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Scalar</source>
-        <translation type="unfinished"></translation>
+        <translation>Scalare</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Array</source>
-        <translation type="unfinished"></translation>
+        <translation>Vettore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Hash</source>
-        <translation type="unfinished"></translation>
+        <translation>Hash</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Symbol table</source>
-        <translation type="unfinished"></translation>
+        <translation>Tabella dei simboli</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Regular expression</source>
-        <translation type="unfinished"></translation>
+        <translation>Espressione regolare</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Substitution</source>
-        <translation type="unfinished"></translation>
+        <translation>Sostituzione</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Backticks</source>
-        <translation type="unfinished"></translation>
+        <translation>Apici inversi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Data section</source>
-        <translation type="unfinished"></translation>
+        <translation>Sezione dati</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Here document delimiter</source>
-        <translation type="unfinished"></translation>
+        <translation>Delimitatore di here document</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Single-quoted here document</source>
-        <translation type="unfinished"></translation>
+        <translation>Here document con apici singoli</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted here document</source>
-        <translation type="unfinished"></translation>
+        <translation>Here document con apici doppi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Backtick here document</source>
-        <translation type="unfinished"></translation>
+        <translation>Here document con apici inversi</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (q)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (q)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qq)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qq)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qx)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qx)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qr)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qr)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qw)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qw)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>POD verbatim</source>
-        <translation type="unfinished"></translation>
+        <translation>POD verbatim</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Subroutine prototype</source>
-        <translation type="unfinished"></translation>
+        <translation>Prototipo subroutine</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Format identifier</source>
-        <translation type="unfinished"></translation>
+        <translation>Formatta identificatore</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Format body</source>
-        <translation type="unfinished"></translation>
+        <translation>Formatta corpo</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted string (interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici doppi (variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Translation</source>
-        <translation type="unfinished"></translation>
+        <translation>Traduzione</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Regular expression (interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Espressione regolare (variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Substitution (interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Sostituzione (variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Backticks (interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Apici inversi (variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Double-quoted here document (interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Here document con apici doppi (variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Backtick here document (interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici inversi (variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qq, interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qq,variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qx, interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qx,variabile interpolata)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Quoted string (qr, interpolated variable)</source>
-        <translation type="unfinished"></translation>
+        <translation>Stringa con apici (qr,variabile interpolata)</translation>
     </message>
 </context>
 <context>
     <name>QsciScintilla</name>
     <message>
         <location filename="../../../qsci/qsciscintilla.cpp" line="+4201"/>
         <source>&amp;Undo</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Annulla</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>&amp;Redo</source>
-        <translation type="unfinished">&amp;Ripeti</translation>
+        <translation>&amp;Ripeti</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Cu&amp;t</source>
-        <translation type="unfinished">&amp;Taglia</translation>
+        <translation>&amp;Taglia</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>&amp;Copy</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Copia</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>&amp;Paste</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Incolla</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Delete</source>
-        <translation type="unfinished">Elimina</translation>
+        <translation>Elimina</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Select All</source>
-        <translation type="unfinished">Seleziona tutto</translation>
+        <translation>Seleziona tutto</translation>
     </message>
 </context>
 <context>
     <name>QtHandles::Figure</name>
     <message>
         <location filename="../graphics/Figure.cc" line="+282"/>
         <source>Figure ToolBar</source>
         <translation>Barra degli strumenti delle figure</translation>
@@ -1054,74 +1054,77 @@ file. Search capabilities may be affecte
 Le funzionalità di ricerca potrebbero risultarne compromesse.
 %1</translation>
     </message>
     <message>
         <location line="+12"/>
         <source>Could not setup the data required for the
 documentation viewer. Only help texts in
 the Command Window will be available.</source>
-        <translation type="unfinished"></translation>
+        <translation>Impossibile initizializzare i dati necessari
+per il visualizzatore della documentazione.
+Soltanto la documentazione nella Finestra
+dei comandi sarà disponibile.</translation>
     </message>
     <message>
         <location line="+21"/>
         <source>Find:</source>
-        <translation type="unfinished"></translation>
+        <translation>Trova:</translation>
     </message>
     <message>
         <location line="+7"/>
         <location line="+1"/>
         <source>Search forward</source>
-        <translation type="unfinished"></translation>
+        <translation>Cerca avanti</translation>
     </message>
     <message>
         <location line="+5"/>
         <location line="+1"/>
         <source>Search backward</source>
-        <translation type="unfinished"></translation>
+        <translation>Cerca indietro</translation>
     </message>
     <message>
         <location line="+74"/>
         <source>Function Index</source>
-        <translation type="unfinished"></translation>
+        <translation>Indice funzioni</translation>
     </message>
     <message>
         <location line="+98"/>
         <source>Go home</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai all&apos;inizio</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Go back</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai indietro</translation>
     </message>
     <message>
         <location line="+14"/>
         <source>Go forward</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai avanti</translation>
     </message>
     <message>
         <location line="+43"/>
         <source>Find</source>
-        <translation type="unfinished">Trova</translation>
+        <translation>Trova</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Zoom in</source>
-        <translation type="unfinished"></translation>
+        <translation>Ingrandisci</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Zoom out</source>
-        <translation type="unfinished"></translation>
+        <translation>Riduci</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Zoom original</source>
-        <translation type="unfinished"></translation>
+        <translation>Dimensione normale</translation>
     </message>
     <message>
         <location line="-206"/>
         <source>Contents</source>
         <translation>Contenuti</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -1132,22 +1135,22 @@ the Command Window will be available.</s
         <location line="+10"/>
         <location line="+39"/>
         <source>Search</source>
         <translation>Cerca</translation>
     </message>
     <message>
         <location line="+84"/>
         <source>Previous pages</source>
-        <translation type="unfinished"></translation>
+        <translation>Pagina precedente</translation>
     </message>
     <message>
         <location line="+14"/>
         <source>Next pages</source>
-        <translation type="unfinished"></translation>
+        <translation>Pagina successiva</translation>
     </message>
     <message>
         <location line="+383"/>
         <source>Unable to register help file %1.</source>
         <translation>Impossibile registrare il file di aiuto %1.</translation>
     </message>
 </context>
 <context>
@@ -1163,37 +1166,37 @@ the Command Window will be available.</s
         <translation>Si faccia riferimento alla documentazione per maggiori informazioni.</translation>
     </message>
 </context>
 <context>
     <name>octave::dw_main_window</name>
     <message>
         <location filename="../src/dw-main-window.cc" line="+49"/>
         <source>&amp;Close</source>
-        <translation type="unfinished">&amp;Chiudi</translation>
+        <translation>&amp;Chiudi</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Close &amp;All</source>
-        <translation type="unfinished"></translation>
+        <translation>Chiudi &amp;tutto</translation>
     </message>
     <message>
         <location line="+6"/>
         <source>Close &amp;Other</source>
-        <translation type="unfinished"></translation>
+        <translation>Chiudi &amp;altro</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Switch to &amp;Left Widget</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai al widget a &amp;sinistra</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Switch to &amp;Right Widget</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai al widget a &amp;destra</translation>
     </message>
 </context>
 <context>
     <name>octave::external_editor_interface</name>
     <message>
         <location filename="../src/external-editor-interface.cc" line="+63"/>
         <location line="+45"/>
         <source>Octave Editor</source>
@@ -1459,17 +1462,17 @@ in scrittura: %2.</translation>
     <message>
         <location line="+5"/>
         <source>Convert Line Endings to &amp;Unix (LF)</source>
         <translation>A capo in stile &amp;Unix (LF)</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Convert Line Endings to Legacy &amp;Mac (CR)</source>
-        <translation type="unfinished">A capo in stile Legacy &amp;Mac (CR)</translation>
+        <translation>A capo in stile Legacy &amp;Mac (CR)</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Navi&amp;gation</source>
         <translation>Navi&amp;gazione</translation>
     </message>
     <message>
         <location line="+4"/>
@@ -1686,17 +1689,18 @@ in scrittura: %2.</translation>
         <location line="+36"/>
         <source>Octave Editor</source>
         <translation>Editor di Octave</translation>
     </message>
     <message>
         <location line="-1977"/>
         <source>Cannot add breakpoint to modified or unnamed file.
 Save and add breakpoint, or cancel?</source>
-        <translation type="unfinished"></translation>
+        <translation>Impossibile aggiungere breakpoint a modificatore o file senza nome.
+Si desidera salvare ed aggiungere un breakpoint o annullare?</translation>
     </message>
     <message>
         <location line="+833"/>
         <source>Goto line</source>
         <translation>Vai alla riga</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -1740,48 +1744,55 @@ sta per essere chiuso ma è stato modificato.  %2</translation>
     </message>
     <message>
         <location line="+126"/>
         <source>There were problems reading the file
 %1
 with the selected encoding %2.
 
 Modifying and saving the file might cause data loss!</source>
-        <translation type="unfinished"></translation>
+        <translation>Si sono riscontrati problemi nel leggere il file
+%1
+con la codifica selezionata %2.
+
+Modicare e salvare potrebbe causare perdita di dati!</translation>
     </message>
     <message>
         <location line="+10"/>
         <source>&amp;Edit anyway</source>
-        <translation type="unfinished"></translation>
+        <translation>&amp;Modifica comunque</translation>
     </message>
     <message>
         <location line="+1"/>
         <location line="+57"/>
         <source>Chan&amp;ge encoding</source>
-        <translation type="unfinished"></translation>
+        <translation>Cam&amp;bia codifica</translation>
     </message>
     <message>
         <location line="-56"/>
         <location line="+49"/>
         <source>&amp;Close</source>
-        <translation type="unfinished">&amp;Chiudi</translation>
+        <translation>&amp;Chiudi</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>Select new default encoding</source>
-        <translation type="unfinished"></translation>
+        <translation>Scegli nuova codifica di default</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Please select a new encoding
 for reloading the current file.
 
 This does not change the default encoding.
 </source>
-        <translation type="unfinished"></translation>
+        <translation>Selezionare una nuova codifica
+per ricaricare il file corrente.
+
+Questo non cambierà la codifica di default.</translation>
     </message>
     <message>
         <location line="+212"/>
         <source>Debug or Save</source>
         <translation>Debug o salvataggio</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -1798,26 +1809,33 @@ Uscire dal debug e salvare?</translation
 %2.</translation>
     </message>
     <message>
         <location line="+182"/>
         <source>The current encoding %1
 can not be applied.
 
 Please select another one!</source>
-        <translation type="unfinished"></translation>
+        <translation>La codifica corrente %1
+non può essere applicata.
+
+Selezionarne un&apos;altra!</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>The current editor contents can not be encoded
 with the selected encoding %1.
 Using it would result in data loss!
 
 Please select another one!</source>
-        <translation type="unfinished"></translation>
+        <translation>I contenuti dell&apos;editor corrente non possono essere
+codificati con la codifica %1.
+Il suo utilizzo comporterebbe perdita di dati!
+
+Selezionarne un&apos;altra!</translation>
     </message>
     <message>
         <location line="-115"/>
         <source>Octave Files (*.m)</source>
         <translation>File di Octave (*.m)</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -1963,17 +1981,17 @@ sia stato rinominato o rimosso. Si desid
         <location line="-381"/>
         <location line="+384"/>
         <source>New Directory...</source>
         <translation>Nuova cartella...</translation>
     </message>
     <message>
         <location line="-329"/>
         <source>Double-click to open file/folder, right click for alternatives</source>
-        <translation type="unfinished"></translation>
+        <translation>Fare doppio click per aprire il file/cartella, o click destro per le alternative.</translation>
     </message>
     <message>
         <location line="+282"/>
         <source>Open</source>
         <translation>Apri</translation>
     </message>
     <message>
         <location line="+5"/>
@@ -2456,17 +2474,17 @@ a:link { text-decoration: underline; col
     <message>
         <location line="+3"/>
         <source>Show filter</source>
         <translation>Mostra filtro</translation>
     </message>
     <message>
         <location line="+121"/>
         <source>Double-click a command to transfer it to the Command Window.</source>
-        <translation type="unfinished"></translation>
+        <translation>Fare doppio click su un comando per trasferirlo alla Finestra dei Comandi.</translation>
     </message>
     <message>
         <location line="+8"/>
         <source>Enter text to filter the command history</source>
         <translation>Inserire del testo per filtrare la cronologia dei comandi</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -2593,17 +2611,18 @@ Fare click su &apos;Successivo&apos; per
 </source>
         <translation>Nome della nuova funzione:
 </translation>
     </message>
     <message>
         <location line="+44"/>
         <source>%1 is a built-in, compiled or inline
 function and can not be edited.</source>
-        <translation type="unfinished"></translation>
+        <translation>%1 è una funzione built-in, compilata o inline
+e non può essere modificata.</translation>
     </message>
     <message>
         <location line="+40"/>
         <source>Can not find function %1</source>
         <translation>Impossibile trovare la funzione %1</translation>
     </message>
     <message>
         <location line="+9"/>
@@ -3090,64 +3109,64 @@ L&apos;inerfaccia grafica verrà chiusa.</translation>
         <translation>Errore critico di Octave</translation>
     </message>
 </context>
 <context>
     <name>octave::settings_dialog</name>
     <message>
         <location filename="../src/settings-dialog.cc" line="-573"/>
         <source>Octave Preferences</source>
-        <translation type="unfinished"></translation>
+        <translation>Preferenze di Octave</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Unable to save preferences.  Missing preferences file or unknown directory.</source>
-        <translation type="unfinished"></translation>
+        <translation>Impossibile salvare le preferenze.  File mancante o cartella sconosciuta.</translation>
     </message>
     <message>
         <location line="+15"/>
         <location line="+4"/>
         <location line="+700"/>
         <source>System setting</source>
-        <translation type="unfinished">Impostazioni di sistema</translation>
+        <translation>Impostazioni di sistema</translation>
     </message>
     <message>
         <location line="-448"/>
         <source>IBeam Cursor</source>
-        <translation type="unfinished">Cursore IBeam</translation>
+        <translation>Cursore IBeam</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Block Cursor</source>
-        <translation type="unfinished">Cursore blocco</translation>
+        <translation>Cursore blocco</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Underline Cursor</source>
-        <translation type="unfinished">Cursore testo sottolineato</translation>
+        <translation>Cursore testo sottolineato</translation>
     </message>
     <message>
         <location line="+164"/>
         <source>Set Octave Startup Directory</source>
-        <translation type="unfinished">Scegli la cartella iniziale di Octave</translation>
+        <translation>Scegli la cartella iniziale di Octave</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Set File Browser Startup Directory</source>
-        <translation type="unfinished">Scegli la cartella iniziale del gestore di file</translation>
+        <translation>Scegli la cartella iniziale del gestore di file</translation>
     </message>
     <message>
         <location line="+503"/>
         <source>Enable attribute colors</source>
-        <translation type="unfinished"></translation>
+        <translation>Attiva colori attibuti</translation>
     </message>
     <message>
         <location line="+3"/>
         <source>Hide tools tips</source>
-        <translation type="unfinished"></translation>
+        <translation>Nascodi suggerimenti strumenti</translation>
     </message>
 </context>
 <context>
     <name>octave::setup_community_news</name>
     <message>
         <location filename="../src/welcome-wizard.cc" line="+42"/>
         <source>Community News</source>
         <translation>Notizie dalla comunità</translation>
@@ -3717,37 +3736,37 @@ Si desidera utilizzare comunque la scorc
     <message>
         <location line="+2"/>
         <source>Show Horizontal Scrollbar</source>
         <translation>Mostra barra di scorrimento orizzontale</translation>
     </message>
     <message>
         <location line="+154"/>
         <source>Documentation Viewer</source>
-        <translation type="unfinished"></translation>
+        <translation>Visualizzatore della documentazione</translation>
     </message>
     <message>
         <location line="+4"/>
         <source>Browser</source>
-        <translation type="unfinished"></translation>
+        <translation>Browser</translation>
     </message>
     <message>
         <location line="+79"/>
         <source>Import shortcuts from file...</source>
-        <translation type="unfinished"></translation>
+        <translation>Importa scorciatoie da file...</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Export shortcuts to file...</source>
-        <translation type="unfinished"></translation>
+        <translation>Esporta scorciatoie su file...</translation>
     </message>
     <message>
         <location line="+11"/>
         <source>Failed to open %1 as Octave shortcut file</source>
-        <translation type="unfinished"></translation>
+        <translation>Impossibile aprire il file %1 come file di scorciatoie di Octave</translation>
     </message>
     <message>
         <location line="-367"/>
         <source>Zoom In</source>
         <translation>Ingrandisci</translation>
     </message>
     <message>
         <location line="+1"/>
@@ -3798,27 +3817,27 @@ Si desidera utilizzare comunque la scorc
     <message>
         <location line="+2"/>
         <source>Document on Keyword</source>
         <translation>Documentazione su parola chiave</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Go to Homepage</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai alla pagina iniziale</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Go Back one Page</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai indietro di una pagina</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Go Forward one Page</source>
-        <translation type="unfinished"></translation>
+        <translation>Vai avanti di una pagina</translation>
     </message>
     <message>
         <location line="+68"/>
         <source>Global</source>
         <translation>Globale</translation>
     </message>
     <message>
         <location line="+3"/>
@@ -3857,22 +3876,22 @@ Si desidera utilizzare comunque la scorc
     <message>
         <location line="+2"/>
         <source>Tab Handling in Dock Widgets</source>
         <translation>Gestione schede nel Dock Widget</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Find &amp; Replace in Dock Widgets</source>
-        <translation type="unfinished"></translation>
+        <translation>Trova e sostituisci nel Dock Widget</translation>
     </message>
     <message>
         <location line="+2"/>
         <source>Zooming in Editor and Documentation</source>
-        <translation type="unfinished"></translation>
+        <translation>Zoom nell&apos;editor e nella documentazione</translation>
     </message>
     <message>
         <location line="+21"/>
         <source>View Menu</source>
         <translation>Menu vista</translation>
     </message>
     <message>
         <location line="+4"/>
@@ -3974,37 +3993,37 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="+1"/>
         <source>Edit variables.</source>
         <translation>Modifica variabili.</translation>
     </message>
     <message>
         <location line="+281"/>
         <source>Foreground</source>
-        <translation type="unfinished"></translation>
+        <translation>Primo piano</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Background</source>
-        <translation type="unfinished">Sfondo</translation>
+        <translation>Sfondo</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Selected Foreground</source>
-        <translation type="unfinished"></translation>
+        <translation>Scegli primo piano</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Selected Background</source>
-        <translation type="unfinished"></translation>
+        <translation>Scegli sfondo</translation>
     </message>
     <message>
         <location line="+1"/>
         <source>Alternate Background</source>
-        <translation type="unfinished"></translation>
+        <translation>Sfondo alternativo</translation>
     </message>
     <message>
         <location line="+273"/>
         <source>Variable Editor Toolbar</source>
         <translation>Barra degli strumenti editor di variabili</translation>
     </message>
     <message>
         <location line="+4"/>
@@ -4349,17 +4368,17 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="+987"/>
         <source>Windows (CRLF)</source>
         <translation>Windows (CRLF)</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Legacy Mac (CR)</source>
-        <translation type="unfinished">Legacy Mac (CR)</translation>
+        <translation>Legacy Mac (CR)</translation>
     </message>
     <message>
         <location line="+5"/>
         <source>Unix (LF)</source>
         <translation>Unix (LF)</translation>
     </message>
     <message>
         <location line="-962"/>
@@ -4557,77 +4576,77 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="-185"/>
         <source>Tab width min.</source>
         <translation>Ampiezza tabulazione min.</translation>
     </message>
     <message>
         <location line="-833"/>
         <source>Preferences</source>
-        <translation type="unfinished">Impostazioni</translation>
+        <translation>Preferenze</translation>
     </message>
     <message>
         <location line="+57"/>
         <source>(requires restart)</source>
-        <translation type="unfinished"></translation>
+        <translation>(richiede riavvio)</translation>
     </message>
     <message>
         <location line="+32"/>
         <source>Use native file dialogs</source>
-        <translation type="unfinished"></translation>
+        <translation>Usa selettore file di sistema</translation>
     </message>
     <message>
         <location line="+53"/>
         <source>Use system icon theme if available (requires restart)</source>
-        <translation type="unfinished"></translation>
+        <translation>Usa tema icone di sistema se disponibile (richiede riavvio)</translation>
     </message>
     <message>
         <location line="+70"/>
         <source>Toolbar Icons</source>
-        <translation type="unfinished"></translation>
+        <translation>Icone barra degli strumenti</translation>
     </message>
     <message>
         <location line="+186"/>
         <source>Language</source>
-        <translation type="unfinished"></translation>
+        <translation>Lingua</translation>
     </message>
     <message>
         <location line="+27"/>
         <source>Style</source>
-        <translation type="unfinished"></translation>
+        <translation>Stile</translation>
     </message>
     <message>
         <location line="+27"/>
         <source>Initial working directory of Octave interpreter</source>
-        <translation type="unfinished"></translation>
+        <translation>Cartella iniziale dell&apos;interprete di Octave</translation>
     </message>
     <message>
         <location line="+18"/>
         <source>Restore last working directory of previous session</source>
-        <translation type="unfinished"></translation>
+        <translation>Ripristina l&apos;ultima cartella di lavoro della sessione precedente</translation>
     </message>
     <message>
         <location line="+45"/>
         <source>Command</source>
-        <translation type="unfinished"></translation>
+        <translation>Comando</translation>
     </message>
     <message>
         <location line="+208"/>
         <source>Set focus to Command Window when running a command from within another widget</source>
-        <translation type="unfinished"></translation>
+        <translation>Seleziona la Finestra dei Comandi quando esegui un comando da un altro widget</translation>
     </message>
     <message>
         <location line="+7"/>
         <source>Print debug location in Command Window in addition to the marker in the editor</source>
-        <translation type="unfinished"></translation>
+        <translation>Stampa punto di debug nella Finestra dei Comandi oltre al marker nell&apos;editor</translation>
     </message>
     <message>
         <location line="+16"/>
         <source>Command Window Colors</source>
-        <translation type="unfinished"></translation>
+        <translation>Colori della Finestra dei Comandi</translation>
     </message>
     <message>
         <location line="+157"/>
         <source>max.</source>
         <translation>max.</translation>
     </message>
     <message>
         <location line="+15"/>
@@ -4732,22 +4751,22 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="+64"/>
         <source>Show completion list automatically</source>
         <translation>Mostra lista di completamento automaticamente</translation>
     </message>
     <message>
         <location line="+34"/>
         <source>Debugging</source>
-        <translation type="unfinished"></translation>
+        <translation>Debug</translation>
     </message>
     <message>
         <location line="+8"/>
         <source>Always show debug breakpoints and pointers (opens related file if closed)</source>
-        <translation type="unfinished"></translation>
+        <translation>Mostra sempre breakpoints di debug e puntatori (apre file corrispondenti se necessario)</translation>
     </message>
     <message>
         <location line="+15"/>
         <source>File handling</source>
         <translation>Gestione file</translation>
     </message>
     <message>
         <location line="+19"/>
@@ -4792,17 +4811,17 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="+22"/>
         <source>Restore last directory of previous session</source>
         <translation>Ripristina l&apos;ultima cartella della sessione precedente</translation>
     </message>
     <message>
         <location line="+32"/>
         <source>Initial file browser directory (only if not synchronized with initial working directory of Octave)</source>
-        <translation type="unfinished"></translation>
+        <translation>Cartella iniziale dell&apos;esploratore file (solo se non sincronizzato con la cartella di lavoro di Octave)</translation>
     </message>
     <message>
         <location line="+36"/>
         <source>Extensions of files to be opened in the default text editor (separated by &quot;;&quot;):</source>
         <translation>Estensioni dei file da aprire nell&apos;editor di testo di default (separate da &quot;;&quot;):</translation>
     </message>
     <message>
         <location line="+39"/>
@@ -4812,17 +4831,17 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="+35"/>
         <source>Colors for variable attributes</source>
         <translation>Colori per gli attributi delle variabili</translation>
     </message>
     <message>
         <location line="+107"/>
         <source>Use Command Window font</source>
-        <translation type="unfinished"></translation>
+        <translation>Usa carattere della Finestra dei Comandi</translation>
     </message>
     <message>
         <location line="+196"/>
         <source>Import shortcut set</source>
         <translation>Importa scorciatoie</translation>
     </message>
     <message>
         <location line="+10"/>
@@ -4908,17 +4927,17 @@ Vorresti salvare le attuali scorciatoie 
     <message>
         <location line="+12"/>
         <source>Use alternating row colors</source>
         <translation>Usa colori alternati per le righe</translation>
     </message>
     <message>
         <location line="+59"/>
         <source>Disable global shortcuts when Command Window has focus</source>
-        <translation type="unfinished"></translation>
+        <translation>Disattiva scorciatoie globali quando la Finestra dei Comandi è attiva</translation>
     </message>
     <message>
         <location line="+211"/>
         <source>HttpProxy</source>
         <translation>HttpProxy</translation>
     </message>
     <message>
         <location line="-2559"/>
diff --git a/libgui/languages/translators b/libgui/languages/translators
--- a/libgui/languages/translators
+++ b/libgui/languages/translators
@@ -7,18 +7,18 @@ be_BY  N.N.
 ca_ES  Salva Ardid <jsardid@gmail.com>
 de_DE  Andreas Weber <andy.weber.aw@googlemail.com>
 es_ES  Valentin Ortega-Clavero <arcanos3030@gmail.com>
        José Luis García Pallero <jgpallero@gmail.com>
 eu_ES  Ander Aurrekoetxea <zaurreko@gmail.com>
 fr_FR  David Bateman <dbateman@free.fr>
        Catalin Codreanu <codreanu.catalin@gmail.com>
        Pantxo Diribarne <pantxo.diribarne@gmail.com>
-it_IT  Massimiliano Fasi <massimiliano.fasi@gmail.com>
-       Marco Caliari <marco.caliari@univr.it>
+it_IT  Marco Caliari <marco.caliari@univr.it>
+       Massimiliano Fasi <massimiliano.fasi@studio.unibo.it>
 ja_JP  Tatsuro Matsuoka <tmacchant@yahoo.co.jp>
 lt_LT  Danius V. <danius-v@inbox.lt>
 nl_NL  Philip Nienhuis <prnienhuis@users.sf.net>
 pt_BR  Felipe G. Nievinski <fgnievinski@gmail.com>
 pt_PT  Carnë Draug <carandraug@octave.org>
 ru_RU  Andriy Shinkarchuck <adriano32.gnu@gmail.com>
 uk_UA  Andriy Shinkarchuck <adriano32.gnu@gmail.com>
 zh_CN  Jeff Bai <aosc@members.fsf.org>
diff --git a/libinterp/parse-tree/jit-typeinfo.cc b/libinterp/parse-tree/jit-typeinfo.cc
--- a/libinterp/parse-tree/jit-typeinfo.cc
+++ b/libinterp/parse-tree/jit-typeinfo.cc
@@ -71,17 +71,17 @@ along with Octave; see the file COPYING.
 #include "ov-builtin.h"
 #include "ov-complex.h"
 #include "ov-scalar.h"
 #include "pager.h"
 #include "interpreter-private.h"
 
 namespace octave
 {
-  static llvm::LLVMContext& context = llvm::getGlobalContext ();
+  static llvm::LLVMContext& context = tree_jit::llvm_context;
 
   std::ostream& jit_print (std::ostream& os, jit_type *atype)
   {
     if (! atype)
       return os << "null";
     return os << atype->name ();
   }
 
diff --git a/libinterp/parse-tree/jit-util.h b/libinterp/parse-tree/jit-util.h
--- a/libinterp/parse-tree/jit-util.h
+++ b/libinterp/parse-tree/jit-util.h
@@ -39,50 +39,54 @@
 
 // we don't want to include llvm headers here, as they require
 // __STDC_LIMIT_MACROS and __STDC_CONSTANT_MACROS be defined in the entire
 // compilation unit
 namespace llvm
 {
   class Value;
   class Module;
-#if defined (LEGACY_PASSMANAGER)
-  namespace legacy
-  {
-    class FunctionPassManager;
-    class PassManager;
-  }
-#else
-  class FunctionPassManager;
-  class PassManager;
-#endif
   class ExecutionEngine;
   class Function;
   class BasicBlock;
   class LLVMContext;
   class Type;
   class StructType;
   class FunctionType;
   class Twine;
   class GlobalValue;
   class GlobalVariable;
   class TerminatorInst;
   class PHINode;
   class TargetMachine;
 
   class ConstantFolder;
 
+#if defined  LLVM_IRBUILDER_HAS_TWO_TEMPLATE_ARGS
+
+  class IRBuilderDefaultInserter;
+
+  template <typename T, typename Inserter>
+  class IRBuilder;
+
+  typedef IRBuilder<ConstantFolder, IRBuilderDefaultInserter>
+    IRBuilderD;
+
+#else
+
   template <bool preserveNames>
   class IRBuilderDefaultInserter;
 
   template <bool preserveNames, typename T, typename Inserter>
   class IRBuilder;
 
   typedef IRBuilder<true, ConstantFolder, IRBuilderDefaultInserter<true>>
     IRBuilderD;
+
+#endif
 }
 
 // some octave classes that are not (yet) in the octave namespace
 class octave_base_value;
 class octave_builtin;
 class octave_value;
 class tree;
 class tree_expression;
diff --git a/libinterp/parse-tree/pt-jit.cc b/libinterp/parse-tree/pt-jit.cc
--- a/libinterp/parse-tree/pt-jit.cc
+++ b/libinterp/parse-tree/pt-jit.cc
@@ -60,17 +60,25 @@ along with Octave; see the file COPYING.
 
 #if defined (HAVE_LLVM_ANALYSIS_BASICALIASANALYSIS_H)
 // In LLVM 3.8.x and later, we use createBasicAAWrapperPass from:
 #  include <llvm/Analysis/BasicAliasAnalysis.h>
 #endif
 // In LLVM 3.7.x and earlier, we use createBasicAliasAnalysisPass
 // from llvm/Analysis/Passes.h (already included above)
 
-#include <llvm/Bitcode/ReaderWriter.h>
+#if defined (HAVE_LLVM_BITCODE_READERWRITER_H)
+// In LLVM <= 3.9, only one header for bitcode read/writer
+#  include <llvm/Bitcode/ReaderWriter.h>
+#else
+// Satrting with LLVM 4.0, two separate headers
+#  include <llvm/Bitcode/BitcodeReader.h>
+#  include <llvm/Bitcode/BitcodeWriter.h>
+#endif
+
 #include <llvm/ExecutionEngine/ExecutionEngine.h>
 // old JIT, LLVM < 3.6.0
 // #include <llvm/ExecutionEngine/JIT.h>
 // MCJIT, LLVM >= 3.0.0
 #include <llvm/ExecutionEngine/MCJIT.h>
 #include "llvm/ExecutionEngine/SectionMemoryManager.h"
 
 #if defined (LEGACY_PASSMANAGER)
@@ -106,29 +114,47 @@ along with Octave; see the file COPYING.
 #  include <llvm/DataLayout.h>
 #else
 #  include <llvm/Target/TargetData.h>
 #endif
 
 #include <llvm/Transforms/IPO.h>
 #include <llvm/Transforms/Scalar.h>
 
+// Starting with LLVM 3.9.0, llvm::createGVNPass has
+// been moved to a new header file named GVN.h
+// (before that it was in llvm/Transforms/Scalar.h)
+#if defined (HAVE_LLVM_TRANSFORMS_SCALAR_GVN_H)
+#  include <llvm/Transforms/Scalar/GVN.h>
+#endif
+
 static bool Vdebug_jit = false;
 
 static bool Vjit_enable = false;
 
 static int Vjit_startcnt = 1000;
 
 static int Vjit_failcnt = 0;
 
 namespace octave
 {
-  static llvm::IRBuilder<> builder (llvm::getGlobalContext ());
-
-  static llvm::LLVMContext& context = llvm::getGlobalContext ();
+  namespace jit
+  {    
+#if defined (LEGACY_PASSMANAGER)
+    typedef llvm::legacy::PassManager PassManager;
+    typedef llvm::legacy::FunctionPassManager FunctionPassManager;
+#else
+    typedef llvm::PassManager PassManager;
+    typedef llvm::FunctionPassManager FunctionPassManager;
+#endif
+  }
+
+  static llvm::IRBuilder<> builder (tree_jit::llvm_context);
+
+  static llvm::LLVMContext& context = tree_jit::llvm_context;
 
   // -------------------- jit_break_exception --------------------
 
   // jit_break is thrown whenever a branch we are converting has only breaks or
   // continues.  This is because all code that follows a break or continue
   // is dead.
   class jit_break_exception : public std::exception
   { };
@@ -1353,21 +1379,23 @@ namespace octave
         builder.SetInsertPoint (m_prelude);
 
         // The jitted function will have only one function argument, of
         // octave_base_value** type
         llvm::Value *arg = &*(m_function->arg_begin ());
 
         for (size_t i = 0; i < m_argument_vec.size (); ++i)
           {
-            // LLVM <= 3.6
-            // llvm::Value *loaded_arg = builder.CreateConstInBoundsGEP1_32 (arg, i);
+#if defined (LLVM_IRBUILDER_CREATECONSTINBOUNDSGEP1_32_REQUIRES_TYPE)
             // LLVM >= 3.7
             llvm::Value *loaded_arg = builder.CreateConstInBoundsGEP1_32 (arg_type, arg, i);
-
+#else
+            // LLVM <= 3.6
+            llvm::Value *loaded_arg = builder.CreateConstInBoundsGEP1_32 (arg, i);
+#endif
             m_arguments[m_argument_vec[i].first] = loaded_arg;
           }
 
         convert (blocks, constants);
       }
     catch (const jit_fail_exception& e)
       {
         m_function->eraseFromParent ();
@@ -2096,16 +2124,18 @@ namespace octave
     return addr;
   }
 
 
   // -------------------- tree_jit --------------------
 
   bool tree_jit::initialized = false;
 
+  llvm::LLVMContext tree_jit::llvm_context;
+
   int tree_jit::next_forloop_number = 0;
   int tree_jit::next_function_number = 0;
   int tree_jit::next_module_number = 0;
 
   tree_jit::tree_jit (void)
     : target_machine (nullptr)
   {
     // target_machine will be truly initialized by tree_jit::do_initialize ()
@@ -2407,18 +2437,22 @@ namespace octave
     if (Vdebug_jit)
       llvm::verifyModule (*m_module);
 
     // DOCUMENT-ME: Why do we need two separate pass managers?
 
     jit::PassManager *module_pass_manager = new jit::PassManager ();
     jit::FunctionPassManager *pass_manager = new jit::FunctionPassManager (m_module);
 
+#if defined (LLVM_HAS_CREATEALWAYSINLINERPASS)
+    // This pass has been removed from LLVM's C++ API after 3.9.0
+    // FIXME: Provide a meaningful replacement instead of simply skipping it?
     module_pass_manager->add (llvm::createAlwaysInlinerPass ());
-
+#endif
+    
     // In 3.6, a pass was inserted in the pipeline to make the DataLayout accessible:
     //    MyPassManager->add(new DataLayoutPass(MyTargetMachine->getDataLayout()));
     // In 3.7, you don’t need a pass, you set the DataLayout on the Module:
     //    MyModule->setDataLayout(MyTargetMachine->createDataLayout());
     //
     // FIXME: autoconf to support <= 3.6
     //
     // #if defined (HAVE_LLVM_DATALAYOUT)
diff --git a/libinterp/parse-tree/pt-jit.h b/libinterp/parse-tree/pt-jit.h
--- a/libinterp/parse-tree/pt-jit.h
+++ b/libinterp/parse-tree/pt-jit.h
@@ -24,36 +24,29 @@ along with Octave; see the file COPYING.
 
 #if ! defined (octave_pt_jit_h)
 #define octave_pt_jit_h 1
 
 #include "octave-config.h"
 
 #if defined (HAVE_LLVM)
 
+#include "jit-util.h"
 #include "jit-typeinfo.h"
 #include "jit-ir.h"
 #include "pt-walk.h"
 #include "symscope.h"
 
 // octave_value_list is not (yet) in the octave namespace
 class octave_value_list;
 
 namespace octave
 {
   namespace jit
   {
-#if defined (LEGACY_PASSMANAGER)
-    typedef llvm::legacy::PassManager PassManager;
-    typedef llvm::legacy::FunctionPassManager FunctionPassManager;
-#else
-    typedef llvm::PassManager PassManager;
-    typedef llvm::FunctionPassManager FunctionPassManager;
-#endif
-
     typedef std::unique_ptr<llvm::Module> ModuleOwner;
     typedef std::unique_ptr<llvm::ExecutionEngine> EngineOwner;
   }
 
   // Convert from the parse tree (AST) to the low level Octave IR.
   class
   jit_convert : public tree_walker
   {
@@ -387,16 +380,22 @@ namespace octave
 
     // ----- Initialization -----
 
   private:
 
     static bool initialized;
     bool do_initialize (void);
 
+    // ----- LLVM context -----
+
+  public:
+
+    static llvm::LLVMContext llvm_context;
+
     // ----- Target machine ----
 
   public:
 
     static const llvm::TargetMachine* get_target_machine (void)
     { return instance ().target_machine; }
 
   private:
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -2604,17 +2604,17 @@ AC_DEFUN([OCTAVE_IEEE754_DATA_FORMAT], [
     ## the one in the octave_ieee_init function in liboctave/lo-ieee.cc.
     AC_MSG_ERROR([IEEE 754 data format required for building Octave])
   fi
 ])
 dnl
 dnl Check for CallInst::addAttribute API
 dnl
 AC_DEFUN([OCTAVE_LLVM_CALLINST_ADDATTRIBUTE_API], [
-  AC_CACHE_CHECK([check LLVM::CallInst::addAttribute arg type is llvm::Attributes],
+  AC_CACHE_CHECK([if llvm::CallInst::addAttribute's arg type is llvm::Attributes],
     [octave_cv_callinst_addattribute_arg_is_attributes],
     [AC_LANG_PUSH(C++)
       AC_COMPILE_IFELSE(
         [AC_LANG_PROGRAM([[
 #if defined (HAVE_LLVM_IR_FUNCTION_H)
           #include <llvm/IR/Instructions.h>
           #include <llvm/IR/Attributes.h>
 #else
@@ -2636,17 +2636,17 @@ AC_DEFUN([OCTAVE_LLVM_CALLINST_ADDATTRIB
     AC_DEFINE(CALLINST_ADDATTRIBUTE_ARG_IS_ATTRIBUTES, 1,
       [Define to 1 if llvm::CallInst:addAttribute arg type is llvm::Attributes.])
   fi
 ])
 dnl
 dnl Check for Function::addAttribute API
 dnl
 AC_DEFUN([OCTAVE_LLVM_FUNCTION_ADDATTRIBUTE_API], [
-  AC_CACHE_CHECK([check llvm::Function::addAttribute arg type is llvm::Attributes],
+  AC_CACHE_CHECK([if llvm::Function::addAttribute's arg type is llvm::Attributes],
     [octave_cv_function_addattribute_arg_is_attributes],
     [AC_LANG_PUSH(C++)
       AC_COMPILE_IFELSE(
         [AC_LANG_PROGRAM([[
 #if defined (HAVE_LLVM_IR_FUNCTION_H)
           #include <llvm/IR/Function.h>
           #include <llvm/IR/Attributes.h>
           #include <llvm/IR/LLVMContext.h>
@@ -2670,17 +2670,17 @@ AC_DEFUN([OCTAVE_LLVM_FUNCTION_ADDATTRIB
     AC_DEFINE(FUNCTION_ADDATTRIBUTE_ARG_IS_ATTRIBUTES, 1,
       [Define to 1 if llvm::Function:addAttribute arg type is llvm::Attributes.])
   fi
 ])
 dnl
 dnl Check for Function::addFnAttr API
 dnl
 AC_DEFUN([OCTAVE_LLVM_FUNCTION_ADDFNATTR_API], [
-  AC_CACHE_CHECK([check LLVM::Function::addFnAttr arg type is llvm::Attributes],
+  AC_CACHE_CHECK([if llvm::Function::addFnAttr's arg type is llvm::Attributes],
     [octave_cv_function_addfnattr_arg_is_attributes],
     [AC_LANG_PUSH(C++)
       AC_COMPILE_IFELSE(
         [AC_LANG_PROGRAM([[
 #if defined (HAVE_LLVM_IR_FUNCTION_H)
           #include <llvm/IR/Function.h>
           #include <llvm/IR/Attributes.h>
 #else
@@ -2699,17 +2699,17 @@ AC_DEFUN([OCTAVE_LLVM_FUNCTION_ADDFNATTR
     AC_DEFINE(FUNCTION_ADDFNATTR_ARG_IS_ATTRIBUTES, 1,
       [Define to 1 if llvm::Function:addFnAttr arg type is llvm::Attributes.])
   fi
 ])
 dnl
 dnl Check for legacy::PassManager API
 dnl
 AC_DEFUN([OCTAVE_LLVM_LEGACY_PASSMANAGER_API], [
-  AC_CACHE_CHECK([check for LLVM::legacy::PassManager],
+  AC_CACHE_CHECK([if llvm::legacy::PassManager exists],
     [octave_cv_legacy_passmanager],
     [AC_LANG_PUSH(C++)
       save_LIBS="$LIBS"
       LIBS="$LLVM_LIBS $LIBS"
       AC_LINK_IFELSE(
         [AC_LANG_PROGRAM([[
           #include <llvm/IR/LegacyPassManager.h>
           ]], [[
@@ -2728,17 +2728,17 @@ AC_DEFUN([OCTAVE_LLVM_LEGACY_PASSMANAGER
     AC_DEFINE(LEGACY_PASSMANAGER, 1,
       [Define to 1 if LLVM::legacy::PassManager exists.])
   fi
 ])
 dnl
 dnl Check for raw_fd_ostream API
 dnl
 AC_DEFUN([OCTAVE_LLVM_RAW_FD_OSTREAM_API], [
-  AC_CACHE_CHECK([check LLVM::raw_fd_ostream arg type is llvm::sys:fs],
+  AC_CACHE_CHECK([if llvm::raw_fd_ostream's arg type is llvm::sys:fs],
     [octave_cv_raw_fd_ostream_arg_is_llvm_sys_fs],
     [AC_LANG_PUSH(C++)
       AC_COMPILE_IFELSE(
         [AC_LANG_PROGRAM([[
           #include <llvm/Support/raw_os_ostream.h>
           ]], [[
           std::string str;
           llvm::raw_fd_ostream fout ("", str, llvm::sys::fs::F_Binary);
@@ -2748,16 +2748,105 @@ AC_DEFUN([OCTAVE_LLVM_RAW_FD_OSTREAM_API
     AC_LANG_POP(C++)
   ])
   if test $octave_cv_raw_fd_ostream_arg_is_llvm_sys_fs = yes; then
     AC_DEFINE(RAW_FD_OSTREAM_ARG_IS_LLVM_SYS_FS, 1,
       [Define to 1 if LLVM::raw_fd_ostream arg type is llvm::sys:fs.])
   fi
 ])
 dnl
+dnl Check llvm::IRBuilder API
+dnl
+AC_DEFUN([OCTAVE_LLVM_IRBUILDER_API], [
+  AC_CACHE_CHECK([if llvm::IRBuilder has two template arguments],
+    [octave_cv_llvm_irbuilder_has_two_template_args],
+    [AC_LANG_PUSH(C++)
+      AC_COMPILE_IFELSE(
+        [AC_LANG_PROGRAM([[
+#if defined (HAVE_LLVM_IR_FUNCTION_H)
+          #include <llvm/IR/LLVMContext.h>
+#else
+          #include <llvm/LLVMContext.h>
+#endif
+#if defined (HAVE_LLVM_IR_IRBUILDER_H)
+          #include <llvm/IR/IRBuilder.h>
+#elif defined (HAVE_LLVM_SUPPORT_IRBUILDER_H)
+          #include <llvm/Support/IRBuilder.h>
+#else
+          #include <llvm/IRBuilder.h>
+#endif
+          using namespace llvm;
+          ]], [[
+          LLVMContext c;
+          IRBuilder<ConstantFolder,IRBuilderDefaultInserter>  irb (c);
+        ]])],
+        octave_cv_llvm_irbuilder_has_two_template_args=yes,
+        octave_cv_llvm_irbuilder_has_two_template_args=no)
+    AC_LANG_POP(C++)
+  ])
+  if test $octave_cv_llvm_irbuilder_has_two_template_args = yes; then
+    AC_DEFINE(LLVM_IRBUILDER_HAS_TWO_TEMPLATE_ARGS, 1,
+      [Define to 1 if llvm::IRBuilder has two template arguments.])
+  fi
+])
+dnl
+dnl Check for llvm::createAlwaysInlinerPass
+dnl
+AC_DEFUN([OCTAVE_LLVM_HAS_CREATEALWAYSINLINERPASS], [
+  AC_CACHE_CHECK([if llvm::createAlwaysInlinerPass exists],
+    [octave_cv_llvm_has_createalwaysinlinerpass],
+    [AC_LANG_PUSH(C++)
+      AC_COMPILE_IFELSE(
+        [AC_LANG_PROGRAM([[
+          #include <llvm/Transforms/IPO.h>
+          ]], [[
+          llvm::Pass *p;
+          p = llvm::createAlwaysInlinerPass ();
+        ]])],
+        octave_cv_llvm_has_createalwaysinlinerpass=yes,
+        octave_cv_llvm_has_createalwaysinlinerpass=no)
+    AC_LANG_POP(C++)
+  ])
+  if test $octave_cv_llvm_has_createalwaysinlinerpass = yes; then
+    AC_DEFINE(LLVM_HAS_CREATEALWAYSINLINERPASS, 1,
+      [Define to 1 if llvm::createAlwaysInlinerPass exists.])
+  fi
+])
+dnl
+dnl Check llvm::IRBuilder::CreateConstInBoundsGEP1_32 API
+dbl
+AC_DEFUN([OCTAVE_LLVM_IRBUILDER_CREATECONSTINBOUNDSGEP1_32_API], [
+  AC_CACHE_CHECK([if llvm::IRBuilder::CreateConstInBoundsGEP1_32 requires a type argument],
+    [octave_cv_llvm_irbuilder_createconstinboundsgep1_32_requires_type],
+    [AC_LANG_PUSH(C++)
+      AC_COMPILE_IFELSE(
+        [AC_LANG_PROGRAM([[
+#if defined (HAVE_LLVM_IR_IRBUILDER_H)
+          #include <llvm/IR/IRBuilder.h>
+#elif defined (HAVE_LLVM_SUPPORT_IRBUILDER_H)
+          #include <llvm/Support/IRBuilder.h>
+#else
+          #include <llvm/IRBuilder.h>
+#endif
+          ]], [[
+          llvm::LLVMContext c;
+          llvm::IRBuilder<>  irb (c);
+          llvm::Value *v;
+          v = irb.CreateConstInBoundsGEP1_32 ((llvm::Value *) nullptr, 0);
+        ]])],
+        octave_cv_llvm_irbuilder_createconstinboundsgep1_32_requires_type=no,
+        octave_cv_llvm_irbuilder_createconstinboundsgep1_32_requires_type=yes)
+    AC_LANG_POP(C++)
+  ])
+  if test $octave_cv_llvm_irbuilder_createconstinboundsgep1_32_requires_type = yes; then
+    AC_DEFINE(LLVM_IRBUILDER_CREATECONSTINBOUNDSGEP1_32_REQUIRES_TYPE, 1,
+      [Define to 1 if llvm::IRBuilder::CreateConstInBoundsGEP1_32 requires a type argument.])
+  fi
+])
+dnl
 dnl OCTAVE_CHECK_FORTRAN_SYMBOL_AND_CALLING_CONVENTIONS
 dnl
 dnl Set variables related to Fortran symbol names (append underscore,
 dnl use uppercase names, etc.) and calling convention (mostly used for
 dnl determining how character strings are passed).
 dnl
 AC_DEFUN([OCTAVE_CHECK_FORTRAN_SYMBOL_AND_CALLING_CONVENTIONS], [
   F77_TOLOWER=yes
